grails-app/views/common/_header.gsp

```diff
@@ -1,5 +1,43 @@
-<%@ page import="org.pih.warehouse.core.RoleType" %>
+<%@ page import="org.pih.warehouse.api.LocalizationApiController; org.pih.warehouse.core.RoleType" %>
 <!-- Block which includes the logo and login banner -->
+<g:if test="${grailsApplication.config.openboxes.helpscout.widget.enabled}">
+    <!--
+      -- Place magical HelpScout incantations early to make
+      -- window.Beacon accessible to any included divs, etc.
+      -->
+    <script type="text/javascript">
+      !function (e, t, n) {
+        function a() {
+          var e = t.getElementsByTagName("script")[0],
+            n = t.createElement("script");
+          n.type = "text/javascript", n.async = !0, n.src = "https://beacon-v2.helpscout.net", e.parentNode.insertBefore(n, e)
+        }
+
+        if (e.Beacon = n = function (t, n, a) {
+          e.Beacon.readyQueue.push({
+            method: t,
+            options: n,
+            data: a
+          })
+        }, n.readyQueue = [], "complete" === t.readyState) return a();
+        e.attachEvent ? e.attachEvent("onload", a) : e.addEventListener("load", a, !1)
+      }(window, document, window.Beacon || function () {
+      });
+      window.Beacon("init", "${grailsApplication.config.openboxes.helpscout.widget.key}")
+      window.Beacon("config", {
+        "color": "${grailsApplication.config.openboxes.helpscout.widget.color}",
+        "enableFabAnimation": false
+      })
+      /*
+       * Use a scriptlet block to prevent html-escaping localization labels, which are JSON.
+       * More modern grails versions have a built-in raw() method to do this, but...
+       * https://stackoverflow.com/questions/1337464/overriding-grails-views-default-codec-html-config-back-to-none
+       */
+      window.Beacon("config", <%=LocalizationApiController.localizeHelpScoutLabels(grailsApplication, session?.user?.locale)%>)
```

Comment:
The setting `grails.views.default.codec = "html"` in _Config.groovy_ html-escapes the json I need to send to `window.Beacon`, which totally breaks the page.

The linked SO article talks about using `<%=STATEMENT%>` to inject data without reformatting it. Localization information will never come from a user so I don't think there are any XSS, etc. issues with doing it in this way, but if there's a better way, I'd be all ears.

Reply:
Could this have been an API request that returns the labels as data?